Reapply "Make functionalization `ViewMeta` serializable with pickle. (#143712)"

Lucaskabela opened the pull request:
### Summary:
NOTE: This is a re-export of https://github.com/pytorch/pytorch/pull/161994 ; the changes between these two PRs is exclusively to the buck/build files 

(Summary from #161994 )
Attempted rebase of https://github.com/pytorch/pytorch/pull/143712. 

This reverts commit 6c713ccb5e0df227dd5b630057cbccd373cbe7d6.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

imported-using-ghimport

Test Plan: Imported from OSS

Differential Revision: D81524507

Pulled By: Lucaskabela


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

Lucaskabela commented:
@pytorchbot label "topic: composability"

dolpm reviewed (approved):
syminteq stuff looks fine. rest has been reviewed prior, so will approve

bdhirsh committed:
Reapply "Make functionalization `ViewMeta` serializable with pickle. (#143712)" (#163769)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/163769

Pull Request resolved: https://github.com/pytorch/pytorch/pull/161994

Attempted rebase of https://github.com/pytorch/pytorch/pull/143712. First I'm sanity checking that OSS CI passes. I'll probably need to abandon this PR and make a diff to check internal CI

This reverts commit 6c713ccb5e0df227dd5b630057cbccd373cbe7d6.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

imported-using-ghimport

Test Plan: Imported from OSS

Reviewed By: dolpm

Differential Revision: D81524507

Pulled By: Lucaskabela

Lucaskabela commented:
@pytorchbot cherry-pick --onto release/2.9 --fixes https://github.com/vllm-project/vllm/issues/24547 -c regression

Lucaskabela commented:
Manual cherry pick: https://github.com/pytorch/pytorch/pull/163873
